packages/cli/README.md

```diff
@@ -127,3 +127,615 @@ jobs:
+    },
+  };
+}
+
+export default {
+  persist: {
+    outputDir: '.code-pushup',
+  },
+  plugins: [await create()],
+  categories: [],
+};
+```
+
+Execute the CLI with `npx code-pushup collect --no-progress` and you should the following output:
+
+**stdout of plugin template**
+
+```sh
+Code PushUp Report - @code-pushup/core@x.y.z
+
+My plugin audits
+● My audit                                                            0
+
+Categories
+┌──────────┬───────┬────────┐
+│ Category │ Score │ Audits │
+└──────────┴───────┴────────┘
+
+Made with ❤ by code-pushup.dev
+```
+
+The CLI argument `--no-progress` is used to get better debugging experience in the console, as the progress bar can
+interfere with debugging logs.
+
+The categories are empty for now. But under the audit listing you can see your plugin title `My plugin` it's listed
+audit `My audit` and the resulting value `0`.
+
+### Plugin Runner
+
+The core of a plugin is defined under the `runner` property and can get implemented in 2 ways:
+
+- as a `RunnerFunction`
+- as a `RunnerConfig`
+
+Your preferred way should be the `RunnerFunction` as it is more flexible and easier to use.
+
+#### RunnerFunction
+
+The `RunnerFunction` is the entry point of your plugin. It is called by the CLI and should return the audit results
+as `AuditOutputs`.
+
+Let's write a real implementation to get familiar with the runner function.
+We will implement a simple file size audit for JavaScript files that tracks the size of specified files in your
+codebase.
+
+Let's start by creating a create function with the correct description for our file size plugin:
+
+<details>
+<summary> <b>file-size plugin setup</b> (collapsed for brevity) </summary>
+
+```typescript
+// code-pushup.config.ts
+import { AuditOutputs, PluginConfig } from '@code-pushup/models';
+import { RunnerFunction } from './plugin-config-runner';
+
+type Options = {
+  directory: string;
+};
+
+async function create(options: Options): Promise<PluginConfig> {
+  const fileSizeAudit = {
+    slug: 'file-size-audit',
+    title: 'File size audit',
+  };
+  return {
+    slug: 'file-size',
+    title: 'File size plugin',
+    icon: 'javascript',
+    audits: [fileSizeAudit],
+    runner: runnerFunction(options),
+  };
+}
+
+// we use a closure to pass options to the runner function for better DX
+async function runnerFunction(options: Options): RunnerFunction {
+  return () => {
+    // implementation follows in the below section
+    return [] as AuditOutputs;
+  };
+}
+
+export default {
+  persist: {
+    outputDir: '.code-pushup',
+  },
+  plugins: [
+    await create({
+      directory: './src',
+    }),
+  ],
+  categories: [],
+};
+```
+
+</details>
+
+The first thing we should think about is how to get the raw data to calculate the metrics.
+In our case we need to get the file name and size of all files in a directory.
+
+The basic implementation looks like this:
+
+**get file-size data**
+
+```typescript
+// code-pushup.config.ts
+
+// ...
```

Comment:
Maybe good to list where the Node APIs are imported from, might not be clear to everyone:

```suggestion
// ...
import { readdir, stat } from 'node:fs/promises';
import { join } from 'node:path';
// ...
```